src/charm.py

```diff
@@ -17,58 +19,50 @@
 )
 
 
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
+
+    def __init__(self, msg, status_type=None):
+        super().__init__()
+
+        self.msg = msg
+        self.status_type = status_type
+        self.status = status_type(msg)
+
+
 class Operator(CharmBase):
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
-        if not self.unit.is_leader():
-            # We can't do anything useful when not the leader, so do nothing.
-            self.model.unit.status = WaitingStatus("Waiting for leadership")
-            return
+
         self.log = logging.getLogger(__name__)
+
         self.profile_image = OCIImageResource(self, "profile-image")
         self.kfam_image = OCIImageResource(self, "kfam-image")
 
-        try:
-            self.interfaces = get_interfaces(self)
-        except NoVersionsListed as err:
-            self.model.unit.status = WaitingStatus(str(err))
-            return
-        except NoCompatibleVersions as err:
-            self.model.unit.status = BlockedStatus(str(err))
-            return
-
         for event in [
             self.on.install,
             self.on.leader_elected,
             self.on.upgrade_charm,
             self.on.config_changed,
+            self.on["kubeflow-profiles"].relation_changed,
         ]:
             self.framework.observe(event, self.main)
 
-        self.framework.observe(
-            self.on["kubeflow-profiles"].relation_changed, self.send_info
-        )
-
-    def send_info(self, event):
-        if self.interfaces["kubeflow-profiles"]:
-            self.interfaces["kubeflow-profiles"].send_data(
-                {
-                    "service-name": self.model.app.name,
-                    "service-port": str(self.model.config["port"]),
-                }
-            )
-
```

Comment:
Same as in Argo.  This feels like an event we can safely keep out of the mono-handler

Reply:
and by argo, I meant [envoy](https://github.com/canonical/envoy-operator/pull/4/files#r755456919) which got resolved to not do this